src/helm/benchmark/adaptation/adapters/language_modeling_adapter.py

```diff
@@ -48,10 +48,20 @@ def generate_requests(self, eval_instance: Instance) -> List[RequestState]:
         """
         max_sequence_length: int = self.window_service.max_sequence_length
         max_request_length: int = self.window_service.max_request_length
-        # Handle max_sequence_and_tokens_length
+        max_sequence_and_generated_tokens_length: int = self.window_service.max_sequence_and_generated_tokens_length
+        max_tokens: int = (
+            min(self.adapter_spec.max_tokens, self.window_service.max_generated_max_tokens)
+            if self.adapter_spec.max_tokens > 0
+            else self.adapter_spec.max_tokens
+        )
+        # Here we assume that there is a max_request_and_generated_tokens_length that we can
```

Comment:
`max_sequence_and_generated_tokens_length` doens't seem to be computed? it's just taken from the `window_service.max_sequence_and_generated_tokens_length`